Implementation

GlebKirsan opened the pull request.

GlebKirsan committed:
step-1

GlebKirsan committed:
feat: add env port usage

GlebKirsan committed:
feat: connect db

GlebKirsan committed:
feat: add nextdate func

GlebKirsan committed:
feat: create task endpoint

GlebKirsan committed:
feat: get all tasks

GlebKirsan committed:
feat: get put task

GlebKirsan committed:
feat: mark done, delete task

GlebKirsan committed:
refactored

GlebKirsan committed:
Merge pull request #1 from GlebKirsan/refactoring

refactored

GlebKirsan committed:
добавить описание

GlebKirsan committed:
fix: json type

GlebKirsan committed:
feat: add authorization

dirtymew commented on cmd/task-tracker/main.go:
```diff
@@ -0,0 +1,69 @@
+
+	"github.com/go-chi/chi/v5"
+	"github.com/pkg/errors"
+
```
что-то с группировкой импортов - нужно поправить везде

используй go imports

GlebKirsan replied:
Поправил.

dirtymew commented on Makefile:
```diff
@@ -0,0 +1,21 @@
+.PHONY:build
+
+run: build
+	go run ./...
```
а почему запускаешь через go run а не собраный бинарник?

GlebKirsan replied:
Поменял, теперь используется собранный бинарник.

dirtymew commented on Makefile:
```diff
@@ -0,0 +1,21 @@
+.DEFAULT_GOAL = build
+
```
make отлично

dirtymew commented on internal/database/database.go:
```diff
@@ -0,0 +1,37 @@
+	log.Println("Database connected")
+
+	log.Println("Running migration")
+	if err = runMigration(db); err != nil {
```
миграции - тоже супер

dirtymew commented on cmd/task-tracker/main.go:
```diff
@@ -0,0 +1,69 @@
+func run() error {
+	cfg := config.Get()
+
+	logger := logger.Get()
```
отлично что добавил логирование

dirtymew commented on internal/database/repositories/task.go:
```diff
@@ -0,0 +1,125 @@
+	return res.LastInsertId()
+}
+
+func (repo *TaskRepo) UpdateTask(task *models.Task) error {
```
название не консистентное - Update

GlebKirsan replied:
Исправил.

dirtymew commented on internal/database/repositories/task.go:
```diff
@@ -0,0 +1,125 @@
+	return err
+}
+
+func (repo *TaskRepo) GetTask(id int64) (*models.Task, error) {
```
Get

dirtymew commented on internal/database/repositories/task.go:
```diff
@@ -0,0 +1,125 @@
+	return tasks, nil
+}
+
+func (repo *TaskRepo) GetAll() ([]models.Task, error) {
```
ну тут я бы упростил и сделал один метод

FInd(filter stirng) - что-то такое

GlebKirsan replied:
Поправил, сделал один метод, но, возможно, не так понял, пришлось перенести логику обработки фильтра в репозиторий.

dirtymew reviewed (changes requested):
Привет Глеб

очень крутая работа - почти нет вопросов
 - логирование
 - make
 - миграции
 - структура проекта
 - архитектура(тут можно было еще интерфесы добавить чтобы связность снизить)
 - документация
 - хороший читаемый код

есть пару мелочей которые стоит поправить

Спасибо получил удовольствие...

ты задание со звездочкой доделывать будешь ?

GlebKirsan committed:
fix: review fixes

- исправлен Makefile, команда run использует результат команды build;
- применён goimports к проекту;
- добавлены интерфейсы;
- названия методов были приведены к одному формату;
- в репозитории добавлены подготовленные запросы.

GlebKirsan committed:
feat: add docker

GlebKirsan commented:
> Привет Глеб
> 
> очень крутая работа - почти нет вопросов
> 
> * логирование
> * make
> * миграции
> * структура проекта
> * архитектура(тут можно было еще интерфесы добавить чтобы связность снизить)
> * документация
> * хороший читаемый код
> 
> есть пару мелочей которые стоит поправить
> 
> Спасибо получил удовольствие...
> 
> ты задание со звездочкой доделывать будешь ?

Большое спасибо! Задание со звёздочкой доделал.

dirtymew commented on internal/database/repositories/task.go:
```diff
@@ -0,0 +1,145 @@
+	var query strings.Builder
+	var args []interface{}
+
+	query.WriteString("SELECT id, date, title, comment, repeat FROM scheduler")
```
можно было вспомогательную функцию обьявить типа getQuery(filter stirng) 
и там из фильтра по switch вернуть уже готовы запрос 

так функцию стораджда разгрузишь

dirtymew commented on internal/database/repositories/task_queries.go:
```diff
@@ -0,0 +1,34 @@
+INSERT INTO scheduler (date, title, comment, repeat) 
+VALUES (:date, :title, :comment, :repeat);`
+
+const update = `
```
не очень удобно лучше по месту запросы писать - ну это на мой взгляд

dirtymew commented on internal/database/repositories/task.go:
```diff
@@ -0,0 +1,145 @@
+			query.WriteString(" WHERE date = :date")
+			args = append(args, sql.Named("date", date.Format("20060102")))
+		} else {
+			query.WriteString(" WHERE title LIKE CONCAT('%', :title, '%')")
```
тут по заданию еще в комменте искать надо - обрати внимание - не буде возвращать за такое работу

dirtymew reviewed (approved):
Привет Глеб

все супер 

Поздравляю и желаю удачи
